File: 0515-find-largest-value-in-each-tree-row/0515-find-largest-value-in-each-tree-row.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> largestValues(TreeNode root) {
        List<Integer>list=new ArrayList<>();
        Queue<TreeNode>que=new LinkedList<>();
        if(root==null){
            return list;
        }
        que.offer(root);
        while(!que.isEmpty()){
            int s=que.size();
            int max=Integer.MIN_VALUE;
            while(s!=0){
                TreeNode temp=que.poll();
                max=Math.max(max,temp.val);
                if(temp.left!=null){
                    que.offer(temp.left);
                }
                if(temp.right!=null){
                    que.offer(temp.right);
                }
                s--;
            }
            list.add(max);
        }
        return list;
    }
}

File: 0515-find-largest-value-in-each-tree-row/0515-find-largest-value-in-each-tree-row.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def largestValues(self, root: Optional[TreeNode]) -> List[int]:
        res=[]
        if root is None:
            return res
        queue=deque([root])
        while queue:
            maxi=-20**8
            size=len(queue)
            for i in range(size):
                node=queue.popleft()
                if node.left:
                    queue.append(node.left)
                if node.right:
                    queue.append(node.right)
                maxi=max(maxi,node.val)
            res.append(maxi)
        return res
                
        